### ISSUE TITLE
Screen reader fails announcing about the group name associated with suggestions available while navigating along the suggestions.

### ISSUE THREAD

**Issue by msftedad** [algolia/docsearch#2041]
## Description
Screen reader fails announcing about the group name associated with suggestions available while navigating along the suggestions.User with visual impairment who rely on screen reader will face difficulties if associated group name is not announced by screen reader.

## Steps to reproduce

1. Open https://www.fast.design/ URL.
2. Navigate to link ‘Documentation’ and invoke it.
3. Navigate to button 'Search' and invoke it.
4. Search for any keyword and navigate to suggestions available. 
5. Listen to screen reader announcement.

**Live reproduction:**

[Associated label is not announced .webm](https://github.com/algolia/docsearch/assets/93735775/79589538-5823-4c1b-a4b4-7f933c05e169)

https://github.com/algolia/docsearch/assets/93735775/5545de2b-09ea-4a42-afbb-43fa8bb56463

## Expected behavior

Screen reader should announce about the group associated with suggestions available while navigating along the suggestions. 

i.e., When user enter the suggestions group, screen reader should announce as '<Group Name>and <suggestion name> along <position set>.

## Environment

OS Ver: 25324.1011
Edge Ver: 112.0.1722.23 (Official build) dev (64-bit)
URL: [FAST](https://www.fast.design/)

**Comment by msftedad** [comment]
Hi Team, Is there any update on this?

**Comment by randombeeper** [comment]
Hi @msftedad , I've never used a screen reader before. Which one are you using or which one do you recommend I use to test this?

**Comment by msftedad** [comment]
@randombeeper, Issue is repro'ing with all the Screen readers Narrator, JAWS, NVDA.Thanks
